pinot-controller/src/main/java/org/apache/pinot/controller/util/CompletionServiceHelper.java

```diff
@@ -129,13 +129,21 @@ private CompletionServiceResponse collectResponse(String tableNameWithType, int
         int statusCode = multiHttpRequestResponse.getResponse().getStatusLine().getStatusCode();
         if (statusCode >= 300) {
           String reason = multiHttpRequestResponse.getResponse().getStatusLine().getReasonPhrase();
-          LOGGER.error("Server: {} returned error: {}, reason: {}", instance, statusCode, reason);
+          LOGGER.error("Server: {} returned error: {}, reason: {} for uri: {}", instance, statusCode, reason, uri);
           completionServiceResponse._failedResponseCount++;
           continue;
         }
         String responseString = EntityUtils.toString(multiHttpRequestResponse.getResponse().getEntity());
-        completionServiceResponse._httpResponses
-            .put(multiRequestPerServer ? uri.toString() : instance, responseString);
+        String key = multiRequestPerServer ? uri.toString() : instance;
+        // If there are multiple requests to the same server with the same URI but different payloads,
+        // we append a count value to the key to ensure each response is uniquely identified.
+        // Otherwise, the map will store only the last response, overwriting previous ones.
+        if (multiRequestPerServer) {
+          int count = completionServiceResponse._instanceToRequestCount.getOrDefault(key, 0) + 1;
+          completionServiceResponse._instanceToRequestCount.put(key, count);
```

Comment:
can do `compute(key, (k, v) -> (v == null) ? 1 : v + 1);` but not a blocking comment